### PR TITLE
Partial views topic update

### LINKED ISSUE #8188: Please give an example of returning a Partial View from a Razor Page method
Born from this issue: #8132 

Currently, I only know how to return a Partial View in a .Net Core 2.1 Razor Pages web app by using a Controller.
I wrote an example project of this here: https://github.com/RyanTaite/DynamicListExample

I would like to know how this can be done from a Razor Page method instead.

Copying from [my comment](https://github.com/aspnet/Docs/issues/8132#issuecomment-414423767) on the above issue:

In `Startup.cs`'s `Configure` method I had to add:
`app.UseMvc(routes => routes.MapRoute("default", "{controller}/{action=Index}/{id?}"));`

Then I add a Controller:

```csharp
using DynamicListExample.Models;
using Microsoft.AspNetCore.Mvc;

namespace DynamicListExample.Controllers
{
    public class ParentController : Controller
    {
        public IActionResult GetChildPartialView()
        {
            return PartialView("_ChildPartialView", new Child());
        }
    }
}
```

And I can call that with a `@Url.Action("GetChildPartialView", "Parent")` from my View/Ajax method

But, I would like to point out that `PartialView("_ChildPartialView", new Child());` throws an error saying:

> Cannot resolve partial view '_ChildPartialView'

It actually can find and return `_ChildPartialView` though. So there will be an error for every `PartialView("_somePartialView", ...)` method call that you just have to ignore

Changing it to `PartialView("Shared/_ChildPartialView", ...)` gets rid of the error in Visual Studio, but fails to locate the View when running the application because it's looking for it in a `View` folder that doesn't exist.
I don't know how to override that behavior.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d670c64c-4030-312c-4355-ee518a23a56a
* Version Independent ID: c20d4db8-d403-9218-c39d-47e56c34e767
* Content: [Partial views in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/partial?view=aspnetcore-2.1)
* Content Source: [aspnetcore/mvc/views/partial.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/views/partial.md)
* Product: **aspnet-core**
* GitHub Login: @ardalis
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8188 

[Internal Review Topic (link will be added when the build completes)](#)

* Expanding topic+sample to include Razor Pages calling partial views.
* Went with one sample, which is currently 2.1. The RP example exactly mirrors the MVC example. This topic is already listed in the table on #5495 that we'll copy and use to track 2.2 sample updates. I've added a note to the table to surface the changes that will be required for 2.2.
* Engineering, plz head-check me on the following remark that I make on Line 157, which offers guidance for streaming the partial vs. not streaming the partial ...

> Since `RenderPartialAsync` streams rendered content, it provides better performance in some scenarios. In performance-critical situations, benchmark the page using both approaches and use the approach that generates a faster response.

... Is that a fair suggestion, and can we can flesh out "some scenarios"?

Thanks to @mikebrind :rocket: for helping with the search locations/order.

### CODE REVIEW

File: aspnetcore/mvc/views/partial.md [hunk 1]
```diff
@@ -1,173 +1,277 @@
...
+::: moniker range=">= aspnetcore-2.1"
+
+A partial view is a *.cshtml* markup file maintained within the *Views* folder (MVC) or *Pages* folder (Razor Pages).
+
+In ASP.NET Core MVC, a controller's <xref:Microsoft.AspNetCore.Mvc.ViewResult> is capable of returning either a view or a partial view. An analogous capability is planned for Razor Pages in ASP.NET Core 2.2. In Razor Pages, a <xref:Microsoft.AspNetCore.Mvc.RazorPages.PageModel>'s <xref:Microsoft.AspNetCore.Mvc.IActionResult> can return a <xref:Microsoft.AspNetCore.Mvc.PartialViewResult>. Referencing and rendering partial views is described in detail in the next section.
 
-As a convention, partial view file names often begin with `_`. This naming convention isn't a requirement, but it helps to visually differentiate partial views from regular views.
+Unlike view or page rendering, a partial view doesn't run *_ViewStart.cshtml*. For more information on *_ViewStart.cshtml*, see <xref:mvc/views/layout>.
+
+Partial view file names often begin with an underscore (`_`). This naming convention isn't required, but it helps to visually differentiate partial views from views and pages.
```
Comment (pranavkm) [inline aspnetcore/mvc/views/partial.md]: Smallish behavior factoid - Razor Pages will ignore any files that start with `_` (even if it includes an `@page` directive)

File: aspnetcore/mvc/views/partial/sample/PartialViewsSample/Pages/DiscoveryRP.cshtml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+@using PartialViewsSample.ViewModels
+@{
+    ViewData["Title"] = "Partial view discovery (Razor Pages)";
+    ViewData["index"] = 0;
+}
+
+<h1>@ViewData["Title"]</h1>
+
+<table class="table table-hover">
+    <caption>Asynchronous partial view discovery examples</caption>
```
Comment (pranavkm) [inline aspnetcore/mvc/views/partial/sample/PartialViewsSample/Pages/DiscoveryRP.cshtml]: Maybe drop the `Asynchronous`? It's an implementation detail that users don't need to know?

**Comment by guardrex** [comment]
Thanks @pranavkm ... Is this remark ok? ...

> Since `RenderPartialAsync` streams rendered content, it provides better performance in some scenarios. In performance-critical situations, benchmark the page using both approaches and use the approach that generates a faster response.

Is there anything else that can be said about the "some scenarios" part? Can I state one (or a few) "for examples" there?

File: aspnetcore/mvc/views/partial.md [hunk 3]
```diff
@@ -1,173 +1,277 @@
...
+::: moniker range=">= aspnetcore-2.1"
+
+A partial view is a *.cshtml* markup file maintained within the *Views* folder (MVC) or *Pages* folder (Razor Pages).
+
+In ASP.NET Core MVC, a controller's <xref:Microsoft.AspNetCore.Mvc.ViewResult> is capable of returning either a view or a partial view. An analogous capability is planned for Razor Pages in ASP.NET Core 2.2. In Razor Pages, a <xref:Microsoft.AspNetCore.Mvc.RazorPages.PageModel>'s <xref:Microsoft.AspNetCore.Mvc.IActionResult> can return a <xref:Microsoft.AspNetCore.Mvc.PartialViewResult>. Referencing and rendering partial views is described in detail in the next section.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: Avoid "next section" in this last sentence, as things will eventually get shuffled around. Instead, link to the section by name.

File: aspnetcore/mvc/views/partial.md [hunk 4]
```diff
@@ -1,173 +1,277 @@
...
 
-As a convention, partial view file names often begin with `_`. This naming convention isn't a requirement, but it helps to visually differentiate partial views from regular views.
+Unlike view or page rendering, a partial view doesn't run *_ViewStart.cshtml*. For more information on *_ViewStart.cshtml*, see <xref:mvc/views/layout>.
+
+Partial view file names often begin with an underscore (`_`). This naming convention isn't required, but it helps to visually differentiate partial views from views and pages. Razor Pages ignores files that start with an underscore, even if the page includes an `@page` directive.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: Clarify that you're talking about MVC in the first sentence.

Is there a way to override this restriction? I can imagine someone, including myself, wanting to avoid renaming partial views just because I'm using Razor Pages.

File: aspnetcore/mvc/views/partial.md [hunk 5]
```diff
@@ -1,173 +1,277 @@
...
+The [Partial Tag Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper) requires ASP.NET Core 2.1 or later.
+
+The Partial Tag Helper renders content asynchronously and uses an HTML-like syntax:
 
-[!code-cshtml[](partial/sample/PartialViewsSample/Views/Home/Discovery.cshtml?name=snippet_PartialTagHelper)]
+```cshtml
+<partial name="_PartialName" />
+```
+
+When a file extension is present, the tag helper references a partial view that must be in the same folder as the markup file calling the partial view:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: tag helper --> Tag Helper

File: aspnetcore/mvc/views/partial.md [hunk 6]
```diff
@@ -1,173 +1,277 @@
...
+Unlike view rendering, a partial view doesn't run *_ViewStart.cshtml*. For more information on *_ViewStart.cshtml*, see <xref:mvc/views/layout>.
+
+Partial view file names often begin with an underscore (`_`). This naming convention isn't required, but it helps to visually differentiate partial views from views.
+
+::: moniker-end
 
 ## Reference a partial view
 
-Within a view page, there are several ways to render a partial view. The best practice is to use asynchronous rendering.
+Within a markup file, there are several ways to reference a partial view. We recommend that apps use one of the asynchronous rendering approaches.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: I know that you're referring to Partial Tag Helpers or Async HTML Helpers. Not everyone will know that. Make this more obvious by calling out the async options in this paragraph. Link to each option here too. That can save some scrolling.

File: aspnetcore/mvc/views/partial.md [hunk 7]
```diff
@@ -1,173 +1,277 @@
...
 
-Replace calls to `@Html.Partial` with `@await Html.PartialAsync` or the Partial Tag Helper. For more information on Partial Tag Helper migration, see [Migrate from an HTML Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper#migrate-from-an-html-helper).
+Replace calls to `@Html.Partial` with `@await Html.PartialAsync` or the [Partial Tag Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper). For more information on Partial Tag Helper migration, see [Migrate from an HTML Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper#migrate-from-an-html-helper).
 
 ::: moniker-end
 
 ## Partial view discovery
 
-When referencing a partial view, you can refer to its location in several ways. For example:
+When a partial view is referenced by name without a file extension, the following locations are searched in the following order:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: Avoid using the word "following" in the same sentence. Maybe something like "the following locations are searched in the order that they appear:".

File: aspnetcore/mvc/views/partial.md [hunk 8]
```diff
@@ -1,173 +1,277 @@
...
 
-```cshtml
-// Uses a view in current folder with this name.
-// If none is found, searches the Shared folder.
-<partial name="_ViewName" />
+**Razor Pages**
 
-// A view with this name must be in the same folder
-<partial name="_ViewName.cshtml" />
+1. Currently executing page's folder
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: I believe the Areas folder is searched too.

File: aspnetcore/mvc/views/partial.md [hunk 9]
```diff
@@ -1,173 +1,277 @@
...
-
-Partial views can be *chained*&mdash;a partial view can call another partial view (as long as you don't create a loop). Within each view or partial view, relative paths are always relative to that view, not to the root or parent view.
+* Different partial views with the same file name are allowed when the partial views are in different folders.
+* When referencing a partial view by name without a file extension and the partial view isn't present in the caller's folder, the partial view in the *Shared* folder supplies the content. Partial views in the *Shared* folder are called *shared partial views* or *default partial views*. This convention for partial view discovery permits override behavior, where the local partial view to the markup file overrides the shared partial view.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: This last sentence should be make clearer. It reads funny to me.

File: aspnetcore/mvc/views/partial.md (hunk 4 above)
Comment (guardrex) [reply aspnetcore/mvc/views/partial.md]: > Is there a way to override this restriction? I can imagine someone, including myself, wanting to avoid renaming partial views just because I'm using Razor Pages.

I don't follow u there. Why would you need to rename partial views with RP?

File: aspnetcore/mvc/views/partial.md (hunk 8 above)
Comment (guardrex) [reply aspnetcore/mvc/views/partial.md]: The framework didn't say it was looking in *Areas* folders. @pranavkm ... anything to note on that?

Comment (pranavkm) [reply aspnetcore/mvc/views/partial.md]: If you're in an area page or area controller, here are the search paths: https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs#L59-L61

File: aspnetcore/mvc/views/partial.md [hunk 10]
```diff
@@ -1,173 +1,280 @@
...
-// Locate the view based on the app root.
-// Paths that start with "/" or "~/" refer to the app root.
-<partial name="~/Views/Folder/_ViewName.cshtml" />
-<partial name="/Views/Folder/_ViewName.cshtml" />
+**Razor Pages**
 
-// Locate the view using a relative path
-<partial name="../Account/_LoginPartial.cshtml" />
-```
+1. Currently executing page's folder
```
Comment (pranavkm) [inline aspnetcore/mvc/views/partial.md]: Sorry, should have caught this - for a Razor Page, this includes the directory graph. `/Pages` would happen to be in it by accident. For instance, doing `<partial name="_Deals" />` from `/Pages/Products/Electronics/TV.cshtml` would look for the partial in:

1. `/_Pages/Produces/Electronics/_Deals.cshtml`
1. `/_Pages/Produces/_Deals.cshtml`
1. `/_Pages/_Deals.cshtml`
1. `/Pages/Shared/_Deals.cshtml`
1. `/Views/Shared/_Deals.cshtml`

https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptions.cs#L125-L129 has the docs for the lookup semantics

Comment (guardrex) [reply aspnetcore/mvc/views/partial.md]: Is the same true for MVC views? ... does it also go up the directory graph?

Comment (pranavkm) [reply aspnetcore/mvc/views/partial.md]: Nope. Just a Razor Pages thing.

Comment (guardrex) [reply aspnetcore/mvc/views/partial.md]: Ok ... I left it that way on the last commit, which has the changes for Pages and the Areas coverage.

File: aspnetcore/mvc/views/partial.md (hunk 4 above)
Comment (scottaddie) [reply aspnetcore/mvc/views/partial.md]: I may have misunderstood the last sentence in that paragraph. I took it to mean that my partial view file name must not begin with underscore if I want to return it from a Razor Pages page model method.

Comment (guardrex) [reply aspnetcore/mvc/views/partial.md]: Yeah ... I see what you mean now. I took "ignores" to just mean that the partial view won't be processed as a Razor Pages page (even if it has the directive), so I need to rephrase it to say exactly that. I'll fix it on the next commit. Thanks for pointing it out. I'm sure it would have been a problem for readers in this format.

File: aspnetcore/mvc/views/partial.md [hunk 11]
```diff
@@ -1,173 +1,286 @@
...
+::: moniker range=">= aspnetcore-2.1"
+
+A partial view is a *.cshtml* markup file maintained within the *Views* folder (MVC) or *Pages* folder (Razor Pages).
+
+In ASP.NET Core MVC, a controller's <xref:Microsoft.AspNetCore.Mvc.ViewResult> is capable of returning either a view or a partial view. An analogous capability is planned for Razor Pages in ASP.NET Core 2.2. In Razor Pages, a <xref:Microsoft.AspNetCore.Mvc.RazorPages.PageModel>'s <xref:Microsoft.AspNetCore.Mvc.IActionResult> can return a <xref:Microsoft.AspNetCore.Mvc.PartialViewResult>. Referencing and rendering partial views is described in the [Reference a partial view](#reference-a-partial-view) section.
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/mvc/views/partial.md]: Already done ... https://github.com/aspnet/Docs/issues/8483

File: aspnetcore/mvc/views/partial.md [hunk 12]
```diff
@@ -1,173 +1,286 @@
...
 
 ::: moniker-end
 
-```cshtml
-// Uses a view in current folder with this name.
-// If none is found, searches the Shared folder.
-@await Html.PartialAsync("_ViewName")
+1. Currently executing view's folder
+1. `/Views/Shared`
+1. `/Pages/Shared`
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: The Pages directory is only searched for 2.0+.

File: aspnetcore/mvc/views/partial.md [hunk 13]
```diff
@@ -1,173 +1,286 @@
...
-@await Html.PartialAsync("_ViewName.cshtml")
+If the partial view is called from an Area view, the following locations are searched:
 
-// Locate the view based on the app root.
-// Paths that start with "/" or "~/" refer to the app root.
-@await Html.PartialAsync("~/Views/Folder/_ViewName.cshtml")
-@await Html.PartialAsync("/Views/Folder/_ViewName.cshtml")
+1. `/Areas/<Area-Name>/Views/<Controller-Name>`
+1. `/Areas/<Area-Name>/Views/Shared`
+1. `/Views/Shared`
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: Based on what I see [here](https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs#L59-L61), these Areas search paths are additions to what you've listed above. They should be added to the list above, to show the search order.

File: aspnetcore/mvc/views/partial.md [hunk 14]
```diff
@@ -1,173 +1,286 @@
...
+*Views/Articles/_ArticleSection.cshtml* is the second partial view referenced by the *Read.cshtml* markup file:
 
 [!code-cshtml[](partial/sample/PartialViewsSample/Views/Articles/_ArticleSection.cshtml)]
 
-At runtime, the partials are rendered into the parent view, which itself is rendered within the shared *_Layout.cshtml*.
+At runtime, the partials are rendered into the parent markup file's rendered output, which itself is rendered within the shared *_Layout.cshtml*. The first partial view renders the article author's name and publication date:
+
+> Abraham Lincoln
+>
+> This partial view from &lt;shared partial view file path from *Views/Shared* (MVC) or *Pages/Shared* (Razor Pages)&gt;.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: You reference Razor Pages here, but that should only display for 2.0+.

File: aspnetcore/toc.md [hunk 15]
```diff
@@ -160,6 +160,7 @@
 ## [Filter methods for Razor Pages](xref:razor-pages/filter)
 ## [Create a Razor Class Library](xref:razor-pages/ui-class)
 ## [Route and app conventions](xref:razor-pages/razor-pages-conventions)
+## [Partial views](xref:mvc/views/partial)
```
Comment (scottaddie) [inline aspnetcore/toc.md]: I recommend removing this ToC link. By putting it here, the Razor Pages ToC node appears for versions prior to 2.0:
image

**Comment by guardrex** [comment]
@scottaddie Still guessing on `/Pages/Shared`, since that isn't in the referenced code.

... and one more commit coming on that list. Stand-by.

**Comment by scottaddie** [comment]
@guardrex Let's have @pranavkm review the search path list for accuracy when you're done.

**Comment by guardrex** [comment]
I don't have knowledge to resolve ...

```csharp
options.AreaViewLocationFormats.Add("/Views/Shared/{0}" + RazorViewEngine.ViewExtension);
```
vs.
```csharp
options.ViewLocationFormats.Add("/Views/Shared/{0}" + RazorViewEngine.ViewExtension);
```
... since they both have the same path.

I may have been headed in the right direction with the first approach that splits the Areas out. Not sure what's best here.

So ..... we have this, which is wrong for MVC (`/Views/Shared` repeated from `AreaViewLocationFormats` and `ViewLocationFormats`) ....

```
::: moniker range=">= aspnetcore-2.1"

**Razor Pages**

1. Currently executing page's folder
1. Directory graph above the page's folder
1. `/Shared`
1. `/Pages/Shared`
1. `/Views/Shared`

**MVC**

::: moniker-end

::: moniker range=">= aspnetcore-2.0"

1. Currently executing view's folder
1. `/Views/Shared`
1. `/Areas/<Area-Name>/Views/<Controller-Name>`
1. `/Areas/<Area-Name>/Views/Shared`
1. `/Views/Shared`
1. `/Pages/Shared`

::: moniker-end

::: moniker range="< aspnetcore-2.0"

1. Currently executing view's folder
1. `/Views/Shared`
1. `/Areas/<Area-Name>/Views/<Controller-Name>`
1. `/Areas/<Area-Name>/Views/Shared`
1. `/Views/Shared`

::: moniker-end
```

... and my original approach was (and I've added "additional" to the Areas locations searched) ...

```
::: moniker range=">= aspnetcore-2.1"

**Razor Pages**

1. Currently executing page's folder
1. Directory graph above the page's folder
1. `/Shared`
1. `/Pages/Shared`
1. `/Views/Shared`

**MVC**

::: moniker-end

::: moniker range=">= aspnetcore-2.0"

1. Currently executing view's folder
1. `/Views/Shared`
1. `/Pages/Shared`

::: moniker-end

::: moniker range="< aspnetcore-2.0"

1. Currently executing view's folder
1. `/Views/Shared`

::: moniker-end

If the partial view is called from an Area view, the following additional locations are searched:

1. `/Areas/<Area-Name>/Views/<Controller-Name>`
1. `/Areas/<Area-Name>/Views/Shared`
1. `/Views/Shared`
```

**Comment by scottaddie** [comment]
@pranavkm What priority is given to Area search paths in relation to the standard search paths? We're trying to determine the correct search order.

**Comment by guardrex** [comment]
Thanks @pranavkm 

Do you have anything else on this one @scottaddie?

File: aspnetcore/mvc/views/partial.md [hunk 16]
```diff
@@ -1,173 +1,293 @@
...
+The following conventions apply to partial view discovery:
 
-Partial views can be *chained*&mdash;a partial view can call another partial view (as long as you don't create a loop). Within each view or partial view, relative paths are always relative to that view, not to the root or parent view.
+* Different partial views with the same file name are allowed when the partial views are in different folders.
+* When referencing a partial view by name without a file extension and the partial view is present in both the caller's folder and the *Shared* folder, the partial view in the caller's folder supplies the partial view. If the partial view isn't present in the caller's folder, the partial view is provided from the *Shared* folder. Partial views in the *Shared* folder are called *shared partial views* or *default partial views*.
+* Partial views can be *chained*&mdash;a partial view can call another partial view if a loop isn't formed by the calls. Relative paths are always relative to the current file, not to the root or parent of the file.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: loop --> circular reference

File: aspnetcore/mvc/views/partial.md [hunk 17]
```diff
@@ -1,173 +1,293 @@
...
+Partial view file names often begin with an underscore (`_`). This naming convention isn't required, but it helps to visually differentiate partial views from views.
+
+::: moniker-end
 
 ## Reference a partial view
 
-Within a view page, there are several ways to render a partial view. The best practice is to use asynchronous rendering.
+Within a markup file, there are several ways to reference a partial view. We recommend that apps use one of the asynchronous rendering approaches:
+
+* [Partial Tag Helper](#partial-tag-helper)
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: The Partial Tag Helper link should only display for 2.1+.

File: aspnetcore/mvc/views/partial.md [hunk 18]
```diff
@@ -1,173 +1,303 @@
...
+Partial view file names often begin with an underscore (`_`). This naming convention isn't required, but it helps to visually differentiate partial views from views.
+
+::: moniker-end
 
 ## Reference a partial view
 
-Within a view page, there are several ways to render a partial view. The best practice is to use asynchronous rendering.
+::: moniker range=">= aspnetcore-2.1"
+
+Within a markup file, there are several ways to reference a partial view. We recommend that apps use one of the asynchronous rendering approaches:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: one of the --> one of the following

File: aspnetcore/mvc/views/partial.md [hunk 19]
```diff
@@ -1,173 +1,303 @@
...
+
+* [Partial Tag Helper](#partial-tag-helper)
+* [Asynchronous HTML Helper](#asynchronous-html-helper)
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.1"
+
+Within a markup file, there are two ways to reference a partial view. We recommend that apps use the [Asynchronous HTML Helper](#asynchronous-html-helper).
```
Comment (scottaddie) [inline aspnetcore/mvc/views/partial.md]: It isn't obvious what the 2nd way is. I know it's the Synchronous HTML Helper option, but that may not be obvious to the reader.
